test/integration/client/cancel-query-tests.js

```diff
@@ -1,5 +1,20 @@
 var helper = require(__dirname+"/test-helper");
 
+test("cancellation of an active query", function() {
+  var client = helper.client();
+
+  var qry = "select pg_sleep(4)";
+  var query1 = client.query(qry);
+  client.cancel(client, query1);
+  var otherClient = helper.client();
```

Comment:
I was a little confused by this behavior, it seems like queries can't run concurrently on a given client? When I tried running this on the same client it would block until the `sleep` finished. If there's a way to do this without opening two connections let me know. 

I'm not 100% certain that the `cancel` will always complete before the `otherClient` issues a query against the database, mostly because I am not sure how `stream.write` works.